src/Packetery/Module/Options/FeatureFlagManager.php

```diff
@@ -221,4 +235,13 @@ public function renderSplitActivationNotice(): void {
 		);
 	}
 
+	/**
+	 * Resets static cache. Used in tests.
+	 *
+	 * @return void
+	 */
+	public static function resetCache(): void {
+		self::$flags = null;
```

Comment:
set up $flags value to NULL in __constructor. 
Then you can drop this method entirely.
You are creating a new instance of the class in tests.

Reply:
That does not work. Static properties keep their value between instances.

Reply:
```
diff --git a/src/Packetery/Module/Options/FeatureFlagManager.php b/src/Packetery/Module/Options/FeatureFlagManager.php
index bc0f44b3..73ad10ab 100644
--- a/src/Packetery/Module/Options/FeatureFlagManager.php
+++ b/src/Packetery/Module/Options/FeatureFlagManager.php
@@ -88,6 +88,7 @@ class FeatureFlagManager {
                $this->optionsProvider = $optionsProvider;
                $this->helper          = $helper;
                $this->wpAdapter       = $wpAdapter;
+               self::$flags = null;
        }
 
        /**
@@ -234,14 +235,4 @@ class FeatureFlagManager {
                        ]
                );
        }
-
-       /**
-        * Resets static cache. Used in tests.
-        *
-        * @return void
-        */
-       public static function resetCache(): void {
-               self::$flags = null;
-       }
-
 }
diff --git a/tests/Module/Options/FeatureFlagManagerTest.php b/tests/Module/Options/FeatureFlagManagerTest.php
index 7ee451ef..3a6b74aa 100644
--- a/tests/Module/Options/FeatureFlagManagerTest.php
+++ b/tests/Module/Options/FeatureFlagManagerTest.php
@@ -79,7 +79,6 @@ class FeatureFlagManagerTest extends TestCase {
                        ->method( 'getTransient' )
                        ->willReturn( $messageDismissed );
 
-               FeatureFlagManager::resetCache();
                $manager = new FeatureFlagManager(
                        $this->createMock( Engine::class ),
                        $optionsProvider,

```